Templates/CSharp/Desktop/MSTestDesktopCSharp.csproj

```diff
@@ -36,17 +36,11 @@
       <IncludeInVSIX>true</IncludeInVSIX>
     </Content>
   </ItemGroup>
-  <!--ItemGroup>
-     <VSTemplate Include="ProjectTemplates\CSharp\Test\MSTestV2Desktop.vstemplate">
-      <SubType>Designer</SubType>
-    </VSTemplate>
-  </ItemGroup-->
-  <!--Import Project="$(TestTemplatesRoot)scripts\build\TestTemplates.targets" /-->
-  <PropertyGroup>
-    <PreBuildEvent>
-      IF EXIST "$(MSBuildThisFileDirectory)ProjectTemplates\CSharp\Test\DesktopUnitTestTemplate.zip" DEL /q /s "$(MSBuildThisFileDirectory)ProjectTemplates\CSharp\Test\DesktopUnitTestTemplate.zip"
-      "$(RepoRoot)tools\ZipItems\ZipItems.exe" "$(MSBuildThisFileDirectory)Desktop.xml" "$(MSBuildThisFileDirectory)\" "$(MSBuildThisFileDirectory)ProjectTemplates\CSharp\Test\DesktopUnitTestTemplate.zip"
-    </PreBuildEvent>
-  </PropertyGroup>
-   <Import Project="$(RepoRoot)eng\Loc.targets" />
+  <Import Project="$(RepoRoot)eng\Loc.targets" />
+  <Target Name="ZipTemplate" BeforeTargets="BeforeBuild">
+        <ZipDirectory
+            SourceDirectory="$(MSBuildThisFileDirectory)ProjectTemplates\CSharp\Test\DesktopUnitTestTemplate"
+            DestinationFile="$(MSBuildThisFileDirectory)ProjectTemplates\CSharp\Test\DesktopUnitTestTemplate.zip"
```

Comment:
Rather than writing to the source directory during the build, it would be better to write the zip file to `$(IntermediateOutputPath)` directory, i.e.  `DestinationFile="$(IntermediateOutputPath)\DesktopUnitTestTemplate.zip"`.

Reply:
The corresponding content item would need to then be added here:

```xml
<Target Name="ZipTemplate" BeforeTargets="BeforeBuild">
<ZipDirectory SourceDirectory="$(MSBuildThisFileDirectory)ProjectTemplates\CSharp\Test\DesktopUnitTestTemplate"
                       DestinationFile="$(IntermediateOutputPath)DesktopUnitTestTemplate.zip" />
<ItemGroup>
  <Content Include="$(IntermediateOutputPath)DesktopUnitTestTemplate.zip" IncludeInVSIX="true" />
</ItemGroup>
```

Reply:
I am not writing back to the source folder, it's actually folder above it.

On the IntermediateOutputPath suggestion, I initially thought about something similar.
Can we retain the structure of folders inside the Vsix ?

Reply:
I think you can specify `VSIXSubPath` on the `Content` item.